src/WooCommerce.php

```diff
@@ -380,7 +385,7 @@ public static function getProductDetailsHtmlDataAttr(\WC_Product $product, $is_d
     $product_id = $product->get_id();
     $product_details = static::getProductDetails($product_id);
 
-    if ('variable' === $product->get_type() && $is_detail_view) {
+    if (('variable' === $product->get_type() && $is_detail_view) || !isset($product_details['variant']) && $product->get_type() === 'variation') {
```

Comment:
The two new conditions need to be wrapped into a parenthesis as you did for the first one, too. This is because and and or have different priorities

Reply:
I would also write the type condition first in both cases (as well as the new lines below), so that it is easier to follow for the reader what the next set of conditions is for.